File: 프로그래머스/lv1/12926. 시저 암호/시저 암호.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s, int n)
{
    string answer = "";

    for (int i = 0; i < s.length(); i++)
    {
        if (s[i] >= 65 && s[i] <= 90)
        {
            if (s[i] + n <= 90)
                answer += s[i] + n;
            else if (s[i] + n > 90)
                answer += 64 + (s[i] + n) - 90;
        }
        else if (s[i] >= 97 && s[i] <= 122)
        {
            if (s[i] + n <= 122)
                answer += s[i] + n;
            else if (s[i] + n > 122)
                answer += 96 + (s[i] + n) - 122;
        }
        else
            answer += " ";
    }
    
    return answer;
}

File: 프로그래머스/1/12926. 시저 암호/시저 암호.java
class Solution {
    public String solution(String s, int n) {
        char[] answer = s.toCharArray();
        
        for (int i = 0; i < answer.length; i++) {
            if (answer[i] == ' ') continue;
            
            char c = (char) ((int) answer[i] + (n % 26));
            if (Character.isUpperCase(answer[i]) && Character.isLowerCase(c) || !Character.isAlphabetic(c)) {
                c -= 26;
            }
            
            answer[i] = c;
        }
        
        return new String(answer);
    }
}